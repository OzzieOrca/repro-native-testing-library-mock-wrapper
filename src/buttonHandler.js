export const buttonHandler = () => {
  console.log('button pressed');
};
