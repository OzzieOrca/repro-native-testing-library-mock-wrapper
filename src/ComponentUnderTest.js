import React from 'react';
import { View, Text } from 'react-native';
import { ButtonWrapper } from './ButtonWrapper';
import { buttonHandler } from './buttonHandler';

export const ComponentUnderTest = () => (
  <View>
    <ButtonWrapper testID="my-button" onPress={buttonHandler}>
      <Text testID="main-content">Main Content</Text>
    </ButtonWrapper>
  </View>
);
