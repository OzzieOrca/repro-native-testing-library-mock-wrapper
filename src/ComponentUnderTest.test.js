import React from 'react';
import { render, fireEvent } from 'native-testing-library';
import {
  render as rntlRender,
  fireEvent as rntlFireEvent,
} from 'react-native-testing-library';

import { ComponentUnderTest } from './ComponentUnderTest';
import { buttonHandler } from './buttonHandler';

jest.mock('./ButtonWrapper', () => ({ ButtonWrapper: 'ButtonWrapper' }));
jest.mock('./buttonHandler', () => ({ buttonHandler: jest.fn() }));

describe('native-testing-library', () => {
  it('should match snapshot', () => {
    const { container } = render(<ComponentUnderTest />);
    expect(container).toMatchSnapshot();
  });

  it('should check main content is correct', () => {
    const { getByTestId } = render(<ComponentUnderTest />);
    expect(getByTestId('main-content').children[0]).toEqual('Main Content');
  });

  it('should press button', () => {
    const { getByTestId } = render(<ComponentUnderTest />);
    // Throws error: Unable to find an element with the testID of: my-button
    fireEvent.press(getByTestId('my-button'));
    expect(buttonHandler).toHaveBeenCalled();
  });
});

describe('react-native-testing-library', () => {
  it('should match snapshot', () => {
    const { toJSON } = rntlRender(<ComponentUnderTest />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should check main content is correct', () => {
    const { getByTestId } = rntlRender(<ComponentUnderTest />);
    expect(getByTestId('main-content').children[0].children[0]).toEqual(
      'Main Content',
    );
  });

  it('should press button', () => {
    const { getByTestId } = rntlRender(<ComponentUnderTest />);
    rntlFireEvent.press(getByTestId('my-button'));
    expect(buttonHandler).toHaveBeenCalled();
  });
});
