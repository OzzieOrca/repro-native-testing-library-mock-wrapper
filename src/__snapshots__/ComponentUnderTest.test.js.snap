// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`native-testing-library should match snapshot 1`] = `
<View>
  <ButtonWrapper
    onPress={[MockFunction]}
    testID="my-button"
  >
    <Text
      testID="main-content"
    >
      Main Content
    </Text>
  </ButtonWrapper>
</View>
`;

exports[`react-native-testing-library should match snapshot 1`] = `
<View>
  <ButtonWrapper
    onPress={[MockFunction]}
    testID="my-button"
  >
    <Text
      testID="main-content"
    >
      Main Content
    </Text>
  </ButtonWrapper>
</View>
`;
